.github/workflows/py-checkstyle.yml

```diff
@@ -16,9 +16,6 @@ name: Python Checkstyle
 on:
   pull_request_target:
     types: [labeled, opened, synchronize, reopened]
-    branches:
-      - main
-      - '0.[0-9]+.[0-9]+'
```

Comment:
What was this used for?

Reply:
this meant that py-checkstyle only ran on the `main` branch and release branches. As I'll be marking the workflow as required, it should run everywhere otherwise we won't be able to merge PRs.

Reply:
Now that I am saying this "out loud" the requirement is only for `main`, so it might have already happened out of the box 🤔

Reply:
I think so since we are merging to `main`. though we may have edge cases where we merge to someone else's PR

Reply:
yup, in that case I still think that it might be good to get some checks. Can add it back if we feel strongly about it